File: carouselpicker/src/main/java/in/goodiebag/carouselpicker/CustomPageTransformer.java
package in.goodiebag.carouselpicker;

import android.content.Context;
import android.view.View;

import androidx.viewpager.widget.ViewPager;

/**
 * Created by pavan on 25/04/17.
 */

public class CustomPageTransformer implements ViewPager.PageTransformer {
    private ViewPager viewPager;
    private CarouselPicker.Mode mode;

    public CustomPageTransformer(Context context, CarouselPicker.Mode mode) {
        this.mode = mode;
    }

    public void transformPage(View view, float position) {
        if (viewPager == null) {
            viewPager = (ViewPager) view.getParent();
        }

        view.setScaleY(1-Math.abs(position));
        view.setScaleX(1-Math.abs(position));

        if (mode == CarouselPicker.Mode.VERTICAL) {
            // Counteract the default slide transition
            view.setTranslationX(view.getWidth() * -position);

            //set Y position to swipe in from top
            float yPosition = position * view.getHeight();
            view.setTranslationY(yPosition);
        }
    }
}

File: carouselpicker/src/main/java/com/goodiebag/carouselpicker/CustomPageTransformer.kt
package com.goodiebag.carouselpicker

import kotlin.math.abs
import android.view.View
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.PageTransformer

/**
 * Created by pavan on 25/04/17.
 */
class CustomPageTransformer(private val mode: CarouselPicker.Mode) :
    PageTransformer {
    private var viewPager: ViewPager? = null
    override fun transformPage(view: View, position: Float) {
        if (viewPager == null) {
            viewPager = view.parent as ViewPager
        }
        view.scaleY = (1 - abs(position.toDouble())).toFloat()
        view.scaleX = (1 - abs(position.toDouble())).toFloat()
        if (mode === CarouselPicker.Mode.VERTICAL) {
            // Counteract the default slide transition
            view.translationX = view.width * -position

            //set Y position to swipe in from top
            val yPosition = position * view.height
            view.translationY = yPosition
        }
    }
}
